interfaces/kalosm-common/src/cache.rs

```diff
@@ -329,7 +346,7 @@ async fn download_into(
 
     let status = response.status();
     if !(status == StatusCode::OK || status == StatusCode::PARTIAL_CONTENT) {
-        bail!("Unexpected status code: {:?}", status);
+        return Err(CacheError::UnexpectedStatusCode(status));
```

Comment:
Could check 401 here and return a separate error. Could also do the same for Partial content.